webscraping/node/course-info/course-info-json-schema.json

```diff
@@ -0,0 +1,58 @@
+{
+    "$schema": "http://json-schema.org/draft-06/schema#",
+
+    "definitions": {
+        "courseCodeString": {
+            "type": "string",
+            "pattern": "^[A-Z]{4} [0-9]{3,4}$"
+        },
+        "arrayOfCourseCodeOrLists": {
+            "type": "array",
+            "items": {
+                "type": "array",
+                "minItems": 1,
+                "items": { "$ref": "#/definitions/courseCodeString" }
+            }
+        },
+        "alphaNumPunctString": {
+            "type": "string",
+            "pattern": "^[\\wñüúáàíÎöóòëéÉèÈ ,'“”‘’\"--­‐‑‒–—/.:;?§()\\[\\]]+$"
+        },
+        "posNumRoundedToHundredths": {
+            "type": "string",
+            "pattern": "^[0-9](\\.[0-9]{1,2})?$"
+        }
+    },
+
+    "title": "Course Info Schema",
+    "description": "A set of courses offered at concordia with information about each one.",
+
+    "type": "array",
```

Comment:
I assume you made this of type array to make the storer faster to write?

Reply:
well the scraper write a json file of the form `[ {course_info_obj}, {course_info_obj}, ... ]` so the schema is written to handle that.  I don't think it's inconvenient for anyone who would want to validate other json they scrape since it's not really that much to just surround their json in square brackets.

Reply:
Yea very easy to handle, just a little weird and possibly misleading from a conceptual standpoint. Not worth changing

Reply:
it would be great if we could put comments in that schema file, but alas we cannot

Reply:
You can't add comments but perhaps the description property can be used for that purpose.